<launch>
    <arg name="namespace" default="$(env UAV_NAMESPACE)"/>
    
    <arg name="depth_topic" default="/blue/camera/depth_registered/points"/>
    <arg name="mask_topic" default="/blue/red_color_filter/mask/compressed"/>

    <node name="pointcloud_filter" type="pointcloud_filter" pkg="pointcloud_filter" ns="$(arg namespace)" output="screen">
        <remap from="/erl_husky/camera/depth_registered/points" to="$(arg depth_topic)"/>
        <remap from="/color_filter/brick_mask/compressed" to="$(arg mask_topic)"/>
        <remap from="/color_filter/nContours" to="red_color_filter/nContours"/>
        <rosparam command="load"  file="$(find pointcloud_filter)/config/brick_distance_parameters.yaml"/>
    </node>
</launch>
